machine.go

```diff
@@ -535,6 +535,11 @@ func (h *Headscale) RefreshMachine(machine *Machine, expiry time.Time) error {
 
 // DeleteMachine softs deletes a Machine from the database.
```

Comment:
What is the difference between hard and soft? why do we have them?

Reply:
Hard actually deletes the entry in the DB, soft just sets the deleted_at field. We do hard when deleting ephemeral nodes.